src/vivarium/framework/values/post_processors.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, Any, Protocol
+
+import pandas as pd
+
+from vivarium.framework.utilities import from_yearly
+from vivarium.framework.values.exceptions import DynamicValueError
+from vivarium.types import NumberLike
+
+if TYPE_CHECKING:
+    from vivarium.framework.values.manager import ValuesManager
+
+
+class PostProcessor(Protocol):
+    def __call__(self, value: Any, manager: ValuesManager) -> Any:
+        ...
+
+
+def rescale_post_processor(value: NumberLike, manager: ValuesManager) -> NumberLike:
+    """Rescales annual rates to time-step appropriate rates.
+
+    This should only be used with a simulation using a
+    :class:`~vivarium.framework.time.DateTimeClock` or another implementation
+    of a clock that traffics in pandas date-time objects.
+
+    Parameters
+    ----------
+    value
+        Annual rates, either as a number or something we can broadcast
+        multiplication over like a :mod:`numpy` array or :mod:`pandas`
+        data frame.
+    manager
+        The ValuesManager for this simulation.
+
+    Returns
+    -------
+        The annual rates rescaled to the size of the current time step size.
+    """
+    if isinstance(value, (pd.Series, pd.DataFrame)):
+        return value.mul(
+            manager.simulant_step_sizes(value.index)
+            .astype("timedelta64[ns]")
+            .dt.total_seconds()
+            / (60 * 60 * 24 * 365.0),
```

Comment:
I thought we used 365.25 for days in a year.

Reply:
Ok I am wrong there is only one instance where we do that here: https://github.com/ihmeuw/vivarium/blob/main/src/vivarium/framework/lookup/table.py#L179 but everywhere else is 365.0